tools/serverpod_cli/lib/src/analyzer/entities/entity_analyzer.dart

```diff
@@ -218,12 +218,22 @@ class SerializableEntityAnalyzer {
     String sourceFileName, [
     ErrorCollector? collector,
   ]) {
-    YamlDocument document = loadYamlDocument(
-      yaml,
-      sourceUrl: Uri.file(sourceFileName),
-      errorListener: collector,
-      recover: true,
-    );
+    YamlDocument document;
+    try {
+      document = loadYamlDocument(
+        yaml,
+        sourceUrl: Uri.file(sourceFileName),
+        errorListener: collector,
+        recover: true,
+      );
+    } catch (error) {
```

Comment:
Suggestion: Use on to catch that specific exception
```suggestion
    } on YamlException catch (error) {
```